### ISSUE TITLE
when running start_macos.sh on macos - nothing happens, just terminal starts

### ISSUE THREAD

**Issue by testter21** [rsxdalv/tts-generation-webui#230]
when running start_macos.sh on macos sonoma - nothing happens, just terminal starts and nothing else is happening there.

**Comment by rsxdalv** [comment]
Hi, thanks for the report. I remember that macs used to have an issue with automating _git_ because of XCode. Have you ever used git from the command line before?

**Comment by testter21** [comment]
Nope, not manually.
So far, I played only with some apps automated via pinokio browser (which works).
But I don't see xcode installed here yet.

**Comment by rsxdalv** [comment]
I see. Could you send me the log/console output?
In the worst case, I might want to see the whole output of the installation from scratch.
Although not everything can be installed on MacOS, I try to make sure that it at least runs.

**Comment by testter21** [comment]
This is my test case with logs
[workflow.zip](https://github.com/rsxdalv/tts-generation-webui/files/13628453/workflow.zip)

**Comment by rsxdalv** [comment]
> This is my test case with logs [workflow.zip](https://github.com/rsxdalv/tts-generation-webui/files/13628453/workflow.zip)
> 
> //update: sound is present, I had wrong output selected on the device
> 
> so apparently, problems were related to subitem.

I examined the logs and they didn't seem to apply to this project. Were you perhaps trying to find text-generation-webui rather than TTS?

**Comment by testter21** [comment]
Oh, sorry, I mixed two separate forums.
I thought someone from there was asking for logs.

Okay, what do you need me to do/send, to replicate the problem?

**Comment by testter21** [comment]
btw, some related info may be found probably here, as this was my starting point:
https://github.com/oobabooga/text-generation-webui/issues/4723

**Comment by testter21** [comment]
Console:

Last login: Sun Dec 10 18:51:35 on ttys000
tester@tester-MBP ~ % 

and that's it.

**Comment by rsxdalv** [comment]
That's interesting, haven't heard of that failure before.

**Comment by testter21** [comment]
I don't know if this is of any help. Current terminal window has name: "zsh" (I think this was installed by pinokio browser?)
I don't know how this works on mac, but perhaps there are different apps for terminal functionalities?

**Comment by rsxdalv** [comment]
Indeed there are different terminals and shells, zsh is one type of shell. I thought someone had used it successfully before but perhaps if you used a different shell it might.

**Comment by rsxdalv** [comment]
> Without guidance it's beyond me right now.
> 
> Besides I don't know whether installing something else as replacement - will not damage other things that use what is installed right now as default apps.

The default should work; installing other things could indeed affect your other projects. You could also try to - open terminal then cd to the directory, then do the `./start_macos.sh` (You can ask ChatGPT about how to do these things).

**Comment by testter21** [comment]
But when i check permissions, local users are set to read/.write on both, folder and file

**Comment by rsxdalv** [comment]
Yes that's normal for a script not to have +x permission. That requires chmod +x command.

**Comment by testter21** [comment]
I'm not programmer per se, nor macos user (migrating from windows), so many things don't tell me too much.
But as you may guess, I'd like to run this thing.

I made comparison between your launching script  that doesn't work here, and oobabooga script that runs normally.
Does it help to figure out what's the issue here?
[compare.xlsx](https://github.com/rsxdalv/tts-generation-webui/files/13649660/compare.xlsx)

Also, is your script isolating dependencies from the system, or propagates, so that other non-isolated apps are affected?

**Comment by rsxdalv** [comment]
This installer is related to oobabooga's installer, as you can see these are not huge differences.
Yes, it installs miniconda and isolates the python and dependencies from the rest.

**Comment by testter21** [comment]
As for rough differences, I see these:

conda deactivation on initialization. I'm guressing this could be the issue.
conda version. not relevant.
environment isolation lines. I'm not sure if this does what I think (isolating dependencies/paths to this project only)
other lines seem to be related to this project only.

**Comment by rsxdalv** [comment]
What I'm saying is - these differences are far too small to be from "working" to "does not even print a single line"
I think you mentioned running oobabooga through another 3rd party app that manages the installation for you, was that right?

**Comment by rsxdalv** [comment]
> I'm not programmer per se, nor macos user (migrating from windows), so many things don't tell me too much. But as you may guess, I'd like to run this thing.
> 
> I made comparison between your launching script that doesn't work here, and oobabooga script that runs normally. Does it help to figure out what's the issue here? [compare.xlsx](https://github.com/rsxdalv/tts-generation-webui/files/13649660/compare.xlsx)
> 
> Also, is your script isolating dependencies from the system, or propagates, so that other non-isolated apps are affected?

UNIX usually does not allow scripts (like .bat files in windows) to be run just like that if you download them from the internet.

**Comment by testter21** [comment]
No, I installed oobabooga according to their page. Downloaded, installed, works, you saw it porobably in the longs that I accidentally posted here.

Prior to that, the only thing I had - was pinokio browser, as I didn't know what to expect on environment isolation for these as scripts.

**Comment by testter21** [comment]
Can it be, that there is some non-visible character (in the path or script), that produces ambiguity on macos?

**Comment by rsxdalv** [comment]
> Can it be, that there is some non-visible character (in the path or script), that produces ambiguity on macos?

To elaborate - that piece of script start_macos.sh is literally an older version of oobabooga's script. They might have fixed something or made it run together with another shell, but I am skeptical; users before have been able to install on a Mac, (although some had the git issue), so I don't see that as being the cause. Up until now I have seen 0 errors.

Did you manage to run the script?

Here's the ChatGPT explanation:

To make a file executable in macOS using the `chmod` command, you need to open the Terminal and use the following command:

```bash
chmod +x start_macos.sh
```

This command grants execute (x) permission to the file `start_macos.sh`. Make sure you navigate to the directory where the file is located or provide the full path if the file is not in the current directory.

Here's a breakdown of the `chmod` command:

- `chmod`: This is the command to change file permissions.
- `+x`: This option adds the execute permission to the file.
- `start_macos.sh`: This is the name of the file you want to modify.

After running this command, you should be able to execute the script by running:

```bash
./start_macos.sh
```

Make sure the script contains valid executable code for it to run successfully. If it's a shell script, the first line should specify the interpreter, such as:

```bash
#!/bin/bash
# rest of your script...
```

Remember to save your changes after editing the file.

**Comment by testter21** [comment]
chmod +x start_macos.sh
in app folder
the same result.

tester@tester-MBP one-click-installers-tts-6.0 % chmod +x start_macos.sh
tester@tester-MBP one-click-installers-tts-6.0 % 

and nothing else.

**Comment by rsxdalv** [comment]
after chmod do
`./start_macos.sh`

**Comment by testter21** [comment]
tester@tester-MBP ~ % cd /Users/tester/Downloads/xtts/one-click-installers-tts/
tester@tester-MBP one-click-installers-tts % chmod +x ./start_macos.sh
tester@tester-MBP one-click-installers-tts %

**Comment by rsxdalv** [comment]
I mean:
```
tester@tester-MBP ~ % cd /Users/tester/Downloads/xtts/one-click-installers-tts/
tester@tester-MBP one-click-installers-tts % chmod +x ./start_macos.sh
tester@tester-MBP one-click-installers-tts %
```
after that 
```
tester@tester-MBP one-click-installers-tts % ./start_macos.sh
```

**Comment by testter21** [comment]
now it started.

**Comment by rsxdalv** [comment]
Please save/post the output in case there are any problems or debug questions.

**Comment by testter21** [comment]
Thanks. I will. So I guess this info has to go somewhere to docs too.

But I wonder, why I could install obaboga hassle-free earlier. Are there some file attributes, to check/uncheck somewhere, so that macos doesn't make such problems in the future?

**Comment by rsxdalv** [comment]
That is a good question. I am not certain that git/downloads page will preserve the +x or that the PC's security will not block it. This is a common pattern on Linux + Mac where you need to chmod +x before running a script, a really common one, although I absolutely wish it was not necessary, I didn't even realize it was. Perhaps other users have a different system configuration and are able to run the file with a double click.
I prefer less documentation and better UX, but I have a huge backlog so I'm not sure if I can afford UX.

**Comment by testter21** [comment]
Okay, as for clean install on macos/sonoma/m-chip, there are some errors.
See the log. 
[log-step1.docx](https://github.com/rsxdalv/tts-generation-webui/files/13650683/log-step1.docx)

**Comment by rsxdalv** [comment]
Ah I have seen that one - the main issue here is that torchaudio library is not loading (the red errors are not important, really). I vaguely remember something about conda incorrectly choosing the Mac OS chip architecture, which I'm pretty sure has been fixed several times before but this could be a new architecture (or a completely different torchaudio error).

**Comment by testter21** [comment]
Ok.
